homeassistant/components/srp_energy/config_flow.py

```diff
@@ -0,0 +1,95 @@
+"""Config flow for SRP Energy."""
+import logging
+
+import voluptuous as vol
+from srpenergy.client import SrpEnergyClient
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_NAME, CONF_PASSWORD, CONF_USERNAME, CONF_ID
+
+from .const import DOMAIN, DEFAULT_NAME  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ID): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
```

Comment:
This is something typical for yaml configuration and not needed with config entries